### PR TITLE
CLN/TYP: remove unused arguments in merge

### PR DESCRIPTION
**Pull request by mzeitlin11**

### CODE REVIEW

File: pandas/core/reshape/merge.py [hunk 1]
```diff
@@ -1479,10 +1474,10 @@ def restore_dropped_levels_multijoin(
...
     right: MultiIndex,
     dropped_level_names,
-    join_index,
-    lindexer,
-    rindexer,
-):
+    join_index: MultiIndex,
+    lindexer: np.ndarray,
+    rindexer: np.ndarray,
+) -> Tuple[List[Index], np.ndarray, List[str]]:
```
Comment (simonjayhawkins) [inline pandas/core/reshape/merge.py:1480]: Can you update the docstring to match for the return type of 'names of combined multiindexes'

File: pandas/core/reshape/merge.py [hunk 2]
```diff
@@ -1583,7 +1578,6 @@ def __init__(
         right_index: bool = False,
         axis: int = 1,
         suffixes: Suffixes = ("_x", "_y"),
-        copy: bool = True,
```
Comment (simonjayhawkins) [inline pandas/core/reshape/merge.py:1586]: I _think_ this should be retained since in the base class `_MergeOperation.__init__`.

**Review by simonjayhawkins — COMMENTED** [review]
Thanks @mzeitlin11 for the PR.

File: pandas/core/reshape/merge.py [hunk 3]
```diff
@@ -979,9 +975,8 @@ def _create_join_index(
         index: Index,
         other_index: Index,
         indexer,
-        other_indexer,
         how: str = "left",
-    ):
+    ) -> np.ndarray:
```
[the remark this answers is not in the document]
Comment (mzeitlin11) [reply pandas/core/reshape/merge.py:979]: Thanks for noticing this, opened #40521 to address

File: pandas/core/reshape/merge.py (hunk 2 above)
Comment (mzeitlin11) [reply pandas/core/reshape/merge.py]: Reverted these changes

File: pandas/core/reshape/merge.py (hunk 1 above)
Comment (mzeitlin11) [reply pandas/core/reshape/merge.py]: Updated docstring. Plan to take a much deeper dive into this code in followups, will try to clean up more inconsistencies once I have a better handle on what's going on here.

**Comment by jreback** [comment]
thanks @mzeitlin11

**Pull request merged**